### PR TITLE
Add build option to toggle libjade implementations in liboqs

### LINKED ISSUE #517: Add option to build against liboqs with OQS_LIBJADE_BUILD=ON
Add an option to set OQS_LIBJADE_BUILD (defaulting to ON) while building oqs-provider to facilitate releases as discussed [here](https://github.com/open-quantum-safe/oqs-provider/discussions/411).

Additionally, check that the correct implementation has been activated.

### PR DESCRIPTION
**Pull request by praveksharma**
This adds a ON/OFF build option OQS_LIBJADE_BUILD which is passed as a environment variable to fullbuild.sh to build liboqs with libjade implementations of algorithms.

Modifies Linux and MacOS CI workflow files to use this build option. Libjade implementations are not presently tested on MacOS in CI; only the STD alg set is tested on MacOS in CI, none of these algs have libjade implementations at the moment.     

Fixes #517.

### CODE REVIEW

File: CONFIGURE.md [hunk 1]
```diff
@@ -186,6 +186,20 @@ facilitated by setting the `liboqs` build option `-DOQS_ALGS_ENABLED=STD` when b
 `liboqs`. The list of algorithms supported by `oqs-provider` is defined by
 the contents of the file `generate.yml` documented in the [pre-build configuration](#pre-build-configuration).
 
+### OQS_LIBJADE_BUILD
+
+This environment variable lets you specify the build option `-DOQS_LIBJADE_BUILD` for building liboqs when using the `fullbuild.sh` script. If this environment variable is not set `fullbuild.sh` defaults to building liboqs with `-DOQS_LIBJADE_BUILD=ON`.
+
+When building liboqs with `-DOQS_LIBJADE_BUILD=ON`, default implementations of post-quantum algorithms are replaced with formally verified implementations sourced from libjade[https://github.com/formosa-crypto/libjade] (if liboqs provied such an implementation.) Please refer to [liboqs documentation](https://github.com/open-quantum-safe/liboqs/blob/main/CONFIGURE.md#oqs_libjade_build) to see which algorithms have formally verified implementations and learn more about the `-DOQS_LIBJADE_BUILD` build option.
```
Comment (SWilson4) [inline CONFIGURE.md:193]:
```suggestion
When building liboqs with `-DOQS_LIBJADE_BUILD=ON`, default implementations of post-quantum algorithms are replaced with formally verified implementations sourced from [libjade](https://github.com/formosa-crypto/libjade) (if liboqs provides such an implementation.) Please refer to [liboqs documentation](https://github.com/open-quantum-safe/liboqs/blob/main/CONFIGURE.md#oqs_libjade_build) to see which algorithms have formally verified implementations and learn more about the `-DOQS_LIBJADE_BUILD` build option.
```
Minor markdown fix.

File: CONFIGURE.md [hunk 2]
```diff
@@ -186,6 +186,20 @@ facilitated by setting the `liboqs` build option `-DOQS_ALGS_ENABLED=STD` when b
...
+
+For example, in order to build liboqs with `-DOQS_LIBJADE_BUILD=OFF`, you might run:
+
+```
+$ env OQS_LIBJADE_BUILD="OFF" bash scripts/fullbuild.sh
```
Comment (SWilson4) [inline CONFIGURE.md:198]: Nit: I always find it annoying when shell code snippets start with $, since it makes them harder to copy/paste.

**Review by SWilson4 — APPROVED** [review]
LGTM barring a couple of nits in documentation. Thanks @praveksharma!

File: CONFIGURE.md (hunk 2 above)
Comment (praveksharma) [reply CONFIGURE.md]: This sounds good. I've edited this and the one other shell snippet in CONFIGURE.md. Thanks for the suggestions @SWilson4!

**Pull request merged**